packages/backend-core/src/users/lookup.ts

```diff
@@ -36,7 +36,7 @@ export async function searchExistingEmails(emails: string[]) {
 // lookup, could be email or userId, either will return a doc
 export async function getPlatformUser(
   identifier: string
-): Promise<PlatformUser | null> {
+): Promise<PlatformUser[] | PlatformUser | null> {
```

Comment:
It returns an array if the following Couch view finds more than one document:

```ts
const viewJs = `function(doc) {
    if (doc.tenantId) {
      emit(doc._id.toLowerCase(), doc._id)
    }

    if (doc.ssoId) {
      emit(doc.ssoId, doc._id)
    }
  }`
```

In this case it is these two documents:

![Screenshot 2024-07-26 at 15 41 23](https://github.com/user-attachments/assets/20d12cf1-f3b8-4a30-b7e0-0300043d50c1)

*The document has the property ssoId, so emit that property as the key* 

![Screenshot 2024-07-26 at 15 41 46](https://github.com/user-attachments/assets/e6f3e5a9-0e6b-467d-8ba1-15e2ed2ba8bd)

*The document has the property tenantId, so emit the _id property as the key*

This will result in two documents matching the SSO ID.

Reply:
If this is the case, we might have all the usages wrong. You are casting the results as `PlatformUser`, so if it returns an array we will have issues. Should we handle the possible arrays on the usages? (fe. `packages/backend-core/src/users/db.ts, line 454`)